Build failure: fish

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

- `fish-4.0.0` has been failing hydra builds on aarch64-linux.
`184 - cargo-test (Failed)`
- This is causing local compilation due to the lack of cache.
- When building on x86-64 using `binfmt.emulatedSystems` for aarch64 three tests are failing.
`85 - noshebang.fish (Failed)`
`156 - complete.py (Failed)`
`179 - torn_escapes.py (Failed)`



### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/292291276

### Relevant log output

```console
[5/8] Building HTML documentation with Sphinx
[6/8] Building man pages with Sphinx
[6/8] Install the project...
-- Install configuration: "Release"
[4/6] cd /build/source-patched/build && env FISH_FORCE_COLOR=1 FISH_SOURCE_DIR=/build/source-patched /nix/store/1an1igpnq15rl2lnnh58c4vzwd9d19xw-cmake-3.31.5/bin/ctest --force-new-ctest-process --output-on-failure --progress
Test project /build/source-patched/build

[...]

        Start  95: random.fish
 84/184 Test  #65: indent.fish ..............................   Passed   17.54 sec
        Start  96: rc-returned.fish
 85/184 Test  #87: nuls.fish ................................   Passed    4.44 sec
        Start  97: read.fish
 86/184 Test  #85: noshebang.fish ...........................***Failed    7.00 sec
Testing file checks/noshebang.fish ... [31mFailure[0m:

  The CHECK on line 33 wants:
    [1m0[0m

  which failed to match line stdout:1:
    [1m127[0m

  Context:
    [31m127[0m <= no check matches
    [1m0[0m
    [31m127[0m <= does not match CHECK on line 34: [34m0[0m
     <= nothing to match CHECK on line 39: [34m0[0m
    [1m0[0m
    [1m126[0m
    [31m127[0m <= does not match CHECK on line 57: [34m126[0m
    [31m127[0m <= no check matches this, previous check on line 57
    [1m0[0m
    [31m127[0m <= does not match CHECK on line 67: [34m0[0m
    [1m126[0m
    [31m127[0m <= does not match CHECK on line 77: [34m126[0m
     <= nothing to match CHECK on line 85: [34m126[0m
    [1m126[0m
    [1mfoo[0m
    [1m0[0m

  when running command:
    ../test/root/bin/fish checks/noshebang.fish

[...]

        Start 177: stdin_nonblocking.py
166/184 Test #170: postexec.py ..............................   Passed   10.18 sec
        Start 178: terminal.py
167/184 Test #171: prompt_redraw_loop.py ....................   Passed    9.62 sec
        Start 179: torn_escapes.py
168/184 Test #169: pipeline.py ..............................   Passed   12.89 sec
        Start 180: tty_ownership.py
169/184 Test #156: complete.py ..............................***Failed   27.75 sec
Testing file pexpects/complete.py:[31mFailed to match pattern:[39m [1m.*still.alive[0m
complete.py:51: [1mtimeout[0m from expect_re(".*still.alive")

[36mEscaped buffer:[0m
\rprompt 3>monster energy=\rprompt 3>monster energy\rprompt 3>monster energ\rprompt 3>monster ener\rprompt 3>monster ene\rprompt 3>monster en\rprompt 3>monster e\rprompt 3>monster \rprompt 3>monster\rprompt 3>monste\rprompt 3>monst\rprompt 3>mons\rprompt 3>mon\rprompt 3>mo\rprompt 3>m\rprompt 3>\r\n⏎                                                                              \r⏎ \r\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\rprompt 4>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 4> \rprompt 4>  \rprompt 4>   \rprompt 4>    \rprompt 4>    # \rprompt 4>    # Make \rprompt 4>    # Make sure \rprompt 4>    # Make sure this \rprompt 4>    # Make sure this function \rprompt 4>    # Make sure this function does \rprompt 4>    # Make sure this function does nothing\rprompt 4>    # Make sure this function does nothing\r\n\x1b]133;C;cmdline_url=%20%20%20%20%23%20Make%20sure%20this%20function%20does%20nothing\x07\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\x1b]133;D;0\x07\x1b[?25h⏎                                                                              \r⏎ \r\rprompt 5>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 5> \rprompt 5>  \rprompt 5>   \rprompt 5>    \rprompt 5>    function \rprompt 5>    function my_is;\rprompt 5>    function my_is; \rprompt 5>    function my_is; :;\rprompt 5>    function my_is; :; \rprompt 5>    function my_is; :; end\rprompt 5>    function my_is; :; end\r\n\x1b]133;C;cmdline_url=%20%20%20%20function%20my_is%3B%20%3A%3B%20end\x07\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\x1b]133;D;0\x07\x1b[?25h⏎                                                                              \r⏎ \r\rprompt 6>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 6> \rprompt 6>  \rprompt 6>   \rprompt 6>    \rprompt 6>    complete \rprompt 6>    complete -c \rprompt 6>    complete -c my_is \rprompt 6>    complete -c my_is -n \rprompt 6>    complete -c my_is -n 'test \rprompt 6>    complete -c my_is -n 'test (count \rprompt 6>    complete -c my_is -n 'test (count (commandline \rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)\rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc))\rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) \rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = \rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' \rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa \rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa arg\rprompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa arg\r\n\x1b]133;C;cmdline_url=%20%20%20%20complete%20-c%20my_is%20-n%20%27test%20%28count%20%28commandline%20-xpc%29%29%20%3D%201%27%20-xa%20arg\x07\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\x1b]133;D;0\x07\x1b[?25h⏎                                                                              \r⏎ \r\rprompt 7>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 7> \rprompt 7>  \rprompt 7>   \rprompt 7>    \rprompt 7>    complete \rprompt 7>    complete -c \rprompt 7>    complete -c my_is \rprompt 7>    complete -c my_is -n \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\n\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\n\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\n\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc)\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\n\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\n\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\n\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n  \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n   \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n    \rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n    )\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n    )'\rprompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\r\nset -l cmd (commandline -xpc) (commandline -ct)\r\nset cmd (string join " " my_is $cmd[3..-1])" "\r\ncommandline --replace --current-process $cmd\r\ncomplete -C"$cmd"\r\n    )'\r\n\x1b]133;C;cmdline_url=%20%20%20%20complete%20-c%20my_is%20-n%20%27__fish_seen_subcommand_from%20not%27%20-xa%20%27%28%0Aset%20-l%20cmd%20%28commandline%20-xpc%29%20%28commandline%20-ct%29%0Aset%20cmd%20%28string%20join%20%22%20%22%20my_is%20%24cmd%5B3..-1%5D%29%22%20%22%0Acommandline%20--replace%20--current-process%20%24cm

[36mWhen written to the tty, this looks like:[0m
[36m<-------[0m

prompt 3>monster energy=
prompt 3>monster energy
prompt 3>monster energ
prompt 3>monster ener
prompt 3>monster ene
prompt 3>monster en
prompt 3>monster e
prompt 3>monster 
prompt 3>monster
prompt 3>monste
prompt 3>monst
prompt 3>mons
prompt 3>mon
prompt 3>mo
prompt 3>m
prompt 3>
⏎                                                                              
⏎ 
[=0u
prompt 4>[=5u
prompt 4> 
prompt 4>  
prompt 4>   
prompt 4>    
prompt 4>    # 
prompt 4>    # Make 
prompt 4>    # Make sure 
prompt 4>    # Make sure this 
prompt 4>    # Make sure this function 
prompt 4>    # Make sure this function does 
prompt 4>    # Make sure this function does nothing
prompt 4>    # Make sure this function does nothing
[=0u[?25h⏎                                                                              
⏎ 

prompt 5>[=5u
prompt 5> 
prompt 5>  
prompt 5>   
prompt 5>    
prompt 5>    function 
prompt 5>    function my_is;
prompt 5>    function my_is; 
prompt 5>    function my_is; :;
prompt 5>    function my_is; :; 
prompt 5>    function my_is; :; end
prompt 5>    function my_is; :; end
[=0u[?25h⏎                                                                              
⏎ 

prompt 6>[=5u
prompt 6> 
prompt 6>  
prompt 6>   
prompt 6>    
prompt 6>    complete 
prompt 6>    complete -c 
prompt 6>    complete -c my_is 
prompt 6>    complete -c my_is -n 
prompt 6>    complete -c my_is -n 'test 
prompt 6>    complete -c my_is -n 'test (count 
prompt 6>    complete -c my_is -n 'test (count (commandline 
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc))
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) 
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' 
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa 
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa arg
prompt 6>    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa arg
[=0u[?25h⏎                                                                              
⏎ 

prompt 7>[=5u
prompt 7> 
prompt 7>  
prompt 7>   
prompt 7>    
prompt 7>    complete 
prompt 7>    complete -c 
prompt 7>    complete -c my_is 
prompt 7>    complete -c my_is -n 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
[=0u
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(

prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
[=5u
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc)
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
[=0u[=5u
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
[=0u[=5u
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
[=0u[=5u
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"

prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
 
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
  
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
   
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
    
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
    )
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
    )'
prompt 7>    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(
set -l cmd (commandline -xpc) (commandline -ct)
set cmd (string join " " my_is $cmd[3..-1])" "
commandline --replace --current-process $cmd
complete -C"$cmd"
    )'
]133;C;cmdline_url=%20%20%20%20complete%20-c%20my_is%20-n%20%27__fish_seen_subcommand_from%20not%27%20-xa%20%27%28%0Aset%20-l%20cmd%20%28commandline%20-xpc%29%20%28commandline%20-ct%29%0Aset%20cmd%20%28string%20join%20%22%20%22%20my_is%20%24cmd%5B3..-1%5D%29%22%20%22%0Acommandline%20--replace%20--current-process%20%24cm[0m
[36m(no trailing newline)[36m------->[0m

Last 10 messages:
OUTPUT     495.70 ms (Line 25): [1mmonster truck ![0m
 INPUT     +18.34 ms (Line 26): [1m\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08[0m
 INPUT     +23.72 ms (Line 27): [1mmonster e[0m
 INPUT    +110.31 ms (Line 29): [1m\t[0m
 INPUT    +108.50 ms (Line 31): [1m![0m
OUTPUT     +14.39 ms (Line 32): [1mmonster energy=![0m
 INPUT     +13.63 ms (Line 33): [1m\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08[0m
 INPUT     +14.40 ms (Line 35): [1m\n    # Make sure this function does nothing\n    function my_is; :; end\n    complete -c my_is -n 'test (count (commandline -xpc)) = 1' -xa arg\n    complete -c my_is -n '__fish_seen_subcommand_from not' -xa '(\n\tset -l cmd (commandline -xpc) (commandline -ct)\n\tset cmd (string join " " my_is $cmd[3..-1])" "\n\tcommandline --replace --current-process $cmd\n\tcomplete -C"$cmd"\n    )'\n\n[0m
 INPUT      +4.18 ms (Line 49): [1mmy_is not \t[0m
 INPUT     +13.56 ms (Line 50): [1mstill.alive[0m

Trying pexpects/complete.py for a second time

[...]

1 test failed

        Start 181: undo.py
170/184 Test #172: read.py ..................................   Passed   11.04 sec
        Start 182: wait.py
171/184 Test #173: set_color.py .............................   Passed    9.52 sec
        Start 183: wildcard_tab.py
172/184 Test #174: sigint.py ................................   Passed    9.40 sec
        Start 184: cargo-test
173/184 Test #176: status.py ................................   Passed    8.91 sec
174/184 Test #177: stdin_nonblocking.py .....................   Passed    9.08 sec
175/184 Test #181: undo.py ..................................   Passed    7.58 sec
176/184 Test #180: tty_ownership.py .........................   Passed    8.39 sec
177/184 Test #178: terminal.py ..............................   Passed   13.96 sec
178/184 Test #183: wildcard_tab.py ..........................   Passed   12.24 sec
179/184 Test #175: signals.py ...............................   Passed   23.89 sec
180/184 Test #182: wait.py ..................................   Passed   17.28 sec
181/184 Test #152: bind_mode_events.py ......................   Passed   50.39 sec
182/184 Test #179: torn_escapes.py ..........................***Failed   27.66 sec
Testing file pexpects/torn_escapes.py:[31mFailed to match pattern:[39m [1mGot\ SIGUSR1\ 3[0m
torn_escapes.py:98: [1mtimeout[0m from expect_str(r"Got SIGUSR1 3")

[36mEscaped buffer:[0m
\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=\x1b[?2004l\x1b[>4;0m\x1b[=0u\x1b>Wacky Handler 3\r\n\x1b[?2004h\x1b[>4;1m\x1b[=5u\x1b=

[36mWhen written to the tty, this looks like:[0m
[36m<-------[0m
[=5u[=0uWacky Handler 3
[=5u[0m
[36m(no trailing newline)[36m------->[0m

Last 10 messages:
 INPUT    1729.39 ms (Line 76): [1m\n[0m
OUTPUT      +7.69 ms (Line 77): [1mprompt 10>[0m
 INPUT      +5.44 ms (Line 81): [1mabc[0m
 INPUT    +105.39 ms (Line 85): [1m\x1bdef[0m
OUTPUT     +11.91 ms (Line 86): [1mGot SIGUSR1 2[0m
OUTPUT      +8.14 ms (Line 87): [1mWacky Handler 2[0m
 INPUT      +5.12 ms (Line 88): [1m\n[0m
OUTPUT      +7.55 ms (Line 89): [1mprompt 11>[0m
 INPUT      +5.04 ms (Line 93): [1mabc\x1b[0m
 INPUT    +104.21 ms (Line 97): [1mdef[0m

Trying pexpects/torn_escapes.py for a second time

[...]

1 test failed

183/184 Test  #15: check-all-fish-files.fish ................   Passed  235.55 sec
184/184 Test #184: cargo-test ...............................   Passed  999.57 sec

98% tests passed, 3 tests failed out of 184

Total Test time (real) = 1119.44 sec

The following tests did not run:
	 62 - git.fish (Skipped)
	 91 - print-help.fish (Skipped)
	124 - tmux-abbr.fish (Skipped)
	125 - tmux-autosuggestion.fish (Skipped)
	126 - tmux-bind.fish (Skipped)
	127 - tmux-bind2.fish (Skipped)
	128 - tmux-commandline.fish (Skipped)
	129 - tmux-complete.fish (Skipped)
	130 - tmux-complete2.fish (Skipped)
	131 - tmux-history-search.fish (Skipped)
	132 - tmux-job.fish (Skipped)
	133 - tmux-multiline-prompt.fish (Skipped)
	134 - tmux-prompt.fish (Skipped)
	135 - tmux-signal.fish (Skipped)
	136 - tmux-transient.fish (Skipped)

The following tests FAILED:
	 85 - noshebang.fish (Failed)
	156 - complete.py (Failed)
	179 - torn_escapes.py (Failed)
Errors while running CTest
FAILED: CMakeFiles/fish_run_tests /build/source-patched/build/CMakeFiles/fish_run_tests 
cd /build/source-patched/build && env FISH_FORCE_COLOR=1 FISH_SOURCE_DIR=/build/source-patched /nix/store/1an1igpnq15rl2lnnh58c4vzwd9d19xw-cmake-3.31.5/bin/ctest --force-new-ctest-process --output-on-failure --progress
ninja: build stopped: subcommand failed.
```

### Additional context

I said hydra can reproduce the build failure but it's technically a different failure.
Hydra is failing on `184 - cargo-test (Failed)`

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.5, NixOS, 25.05 (Warbler), 25.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@adamcstephens 
@cole-h 
@winterqt 
@SigmaSquadron 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fi/fish/1d78c8bd4295262a3118f478e6b3a7c7536fa282.patch
+++ b/pkgs/by-name/fi/fish/1d78c8bd4295262a3118f478e6b3a7c7536fa282.patch
@@ -1,0 +1,74 @@
+From 1d78c8bd4295262a3118f478e6b3a7c7536fa282 Mon Sep 17 00:00:00 2001
+From: Johannes Altmanninger <aclopte@gmail.com>
+Date: Wed, 19 Mar 2025 09:39:04 +0100
+Subject: [PATCH] Fix concurrent setlocale() in string escape tests
+
+In our C++ implementation, these tests were run serially.  As pointed out in
+https://github.com/fish-shell/fish-shell/issues/11254#issuecomment-2735623229
+we run them in parallel now, which means that one test could be changing
+the global locale used by another.
+
+In theory this could be fine because all tests are setting setting the
+global locale to the same thing but the existence of a lock suggests that
+setlocale() is not guaranteed to be atomic, so it's possible that another
+thread uses a temporarily-invalid locale.
+
+Fixes #11254
+---
+ src/tests/string_escape.rs | 14 ++++++++------
+ 1 file changed, 8 insertions(+), 6 deletions(-)
+
+diff --git a/src/tests/string_escape.rs b/src/tests/string_escape.rs
+index ba8ee7534ebf..4428d679cd35 100644
+--- a/src/tests/string_escape.rs
++++ b/src/tests/string_escape.rs
+@@ -1,3 +1,5 @@
++use std::sync::MutexGuard;
++
+ use crate::common::{
+     escape_string, str2wcstring, unescape_string, wcs2string, EscapeFlags, EscapeStringStyle,
+     UnescapeStringStyle, ENCODE_DIRECT_BASE, ENCODE_DIRECT_END,
+@@ -10,21 +12,21 @@ use rand::{Rng, RngCore};
+ 
+ /// wcs2string is locale-dependent, so ensure we have a multibyte locale
+ /// before using it in a test.
+-fn setlocale() {
+-    let _guard = LOCALE_LOCK.lock().unwrap();
++fn setlocale() -> MutexGuard<'static, ()> {
++    let guard = LOCALE_LOCK.lock().unwrap();
+ 
+     #[rustfmt::skip]
+     const UTF8_LOCALES: &[&str] = &[
+         "C.UTF-8", "en_US.UTF-8", "en_GB.UTF-8", "de_DE.UTF-8", "C.utf8", "UTF-8",
+     ];
+     if crate::libc::MB_CUR_MAX() > 1 {
+-        return;
++        return guard;
+     }
+     for locale in UTF8_LOCALES {
+         let locale = std::ffi::CString::new(locale.to_owned()).unwrap();
+         unsafe { libc::setlocale(libc::LC_CTYPE, locale.as_ptr()) };
+         if crate::libc::MB_CUR_MAX() > 1 {
+-            return;
++            return guard;
+         }
+     }
+     panic!("No UTF-8 locale found");
+@@ -100,7 +102,7 @@ fn test_escape_var() {
+ }
+ 
+ fn escape_test(escape_style: EscapeStringStyle, unescape_style: UnescapeStringStyle) {
+-    setlocale();
++    let _locale_guard = setlocale();
+     let seed: u128 = 92348567983274852905629743984572;
+     let mut rng = get_seeded_rng(seed);
+ 
+@@ -174,7 +176,7 @@ fn str2hex(input: &[u8]) -> String {
+ /// string comes back through double conversion.
+ #[test]
+ fn test_convert() {
+-    setlocale();
++    let _locale_guard = setlocale();
+     let seed = get_rng_seed();
+     let mut rng = get_seeded_rng(seed);
+     let mut origin = Vec::new();

--- a/pkgs/by-name/fi/fish/package.nix
+++ b/pkgs/by-name/fi/fish/package.nix
@@ -188,6 +188,10 @@ stdenv.mkDerivation (finalAttrs: {
     # * <https://github.com/LnL7/nix-darwin/issues/122>
     # * <https://github.com/fish-shell/fish-shell/issues/7142>
     ./nix-darwin-path.patch
+
+    # remove 4.0.2
+    # https://github.com/fish-shell/fish-shell/issues/11254
+    ./1d78c8bd4295262a3118f478e6b3a7c7536fa282.patch
   ];
 
   # Fix FHS paths in tests
@@ -317,16 +321,12 @@ stdenv.mkDerivation (finalAttrs: {
       darwin.system_cmds
     ];
 
+  # we target the top-level make target which runs all the cmake/ctest
+  # tests, including test_cargo-test
   checkTarget = "fish_run_tests";
   preCheck = ''
     export TERMINFO="${ncurses}/share/terminfo"
   '';
-  checkFlags = lib.optionals (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isAarch64) [
-    # thread 'tests::string_escape::test_escape_random_url' panicked at src/tests/string_escape.rs:122:9:
-    # assertion `left == right` failed: Escaped and then unescaped string ... but got back a different string ...
-    # https://github.com/fish-shell/fish-shell/issues/11254
-    "--skip=tests::string_escape::test_escape_random_url"
-  ];
 
   nativeInstallCheckInputs = [
     versionCheckHook

